VariantShredding.md

```diff
@@ -25,290 +25,325 @@
-For each value in the data, at most one of the `typed_value` and `variant_value` may be non-null.
-A writer may omit either field, which is equivalent to all rows being null.
+An object is _partially shredded_ when the `value` is an object and the `typed_value` is a shredded object.
+Writers must not produce data where both `value` and `typed_value` are non-null, unless the Variant value is an object.
 
-Dictionary IDs in a `variant_value` field refer to entries in the top-level `metadata` field.
+If a Variant is missing in a context where a value is required, readers must return a Variant null (`00`): basic type 0 (primitive) and physical type 0 (null).
+For example, if a Variant is required (like `measurement` above) and both `value` and `typed_value` are null, the returned `value` must be `00` (Variant null).
 
-For an `object`, a null field means that the field does not exist in the reconstructed Variant object.
-All elements of an `array` must be non-null, since array elements cannote be missing.
+### Shredded Value Types
 
-| typed_value | variant_value | Meaning |
-|-------------|----------------|---------|
-| null | null | Field is Variant Null (not missing) in the reconstructed Variant. |
-| null | non-null | Field may be any type in the reconstructed Variant. |
-| non-null | null | Field has this column’s type in the reconstructed Variant. |
-| non-null | non-null | Invalid |
+Shredded values must use the following Parquet types:
 
-The `typed_value` may be absent from the Parquet schema for any field, which is equivalent to its value being always null (in which case the shredded field is always stored as a Variant binary).
-By the same token, `variant_value` may be absent, which is equivalent to their value being always null (in which case the field will always have the value Null or have the type of the `typed_value` column).
+| Variant Type                | Equivalent Parquet Type           |
+|-----------------------------|-----------------------------------|
+| boolean                     | BOOLEAN                           |
+| int8                        | INT(8, signed=true)               |
+| int16                       | INT(16, signed=true)              |
+| int32                       | INT32 / INT(32, signed=true)      |
+| int64                       | INT64 / INT(64, signed=true)      |
+| float                       | FLOAT                             |
+| double                      | DOUBLE                            |
+| decimal4                    | INT32 / DECIMAL(precision, scale) |
+| decimal8                    | INT64 / DECIMAL(precision, scale) |
+| decimal16                   | DECIMAL(precision, scale)         |
+| date                        | DATE                              |
+| time                        | TIME(false, MICROS)               |
+| timestamp with time zone    | TIMESTAMP(true, MICROS|NANOS)     |
+| timestamp without time zone | TIMESTAMP(false, MICROS|NANOS)    |
```

Comment:
nit: The code seems correct but the preview cuts off the words somehow.

![image](https://github.com/user-attachments/assets/b423f903-502f-4d0a-a1ec-701353bbcc16)

Reply:
I fixed this by adding separate columns for physical type and logical type annotation.